### DISCUSSION TITLE
Wrong Database Size Calculation for Free Plan

### DISCUSSION THREAD

**Discussion by mertbulan** [supabase/supabase#29681] [Questions] [answered]
Hello,

I've been using supabase for the last couple of months and quite happy with the service. I'm currently using the free plan and trying to keep my database size under the limit. Right now my database size is around 488MB and everyday I'm inserting new records, removing the same amount of records and then vacuum the table. Therefore, I would expect the database size to stay under the limit. On the other hand, when I look at the dashboard (Account Settings > Usage), it shows that the average database size is 868MB. I don't know where that number comes from.

<img width="577" alt="Screenshot 2024-10-04 at 10 19 34" src="https://github.com/user-attachments/assets/f28e8064-c149-48c8-acc5-4e835ce9eb46">

I use this SQL query to check the database size and I see that it is 488MB:

```sql
select
  sum(pg_database_size(pg_database.datname)) / (1024 * 1024) as db_size_mb
from pg_database;
```

When I check the Reports page for the project, I see the correct database size there:

<img width="403" alt="Screenshot 2024-10-04 at 10 27 40" src="https://github.com/user-attachments/assets/7f9de0d6-56f5-493c-bc80-8623ad81c634">

I'm wondering why there is a huge difference between the Reports page and the Usage page because I don't want my database to become read-only since I'm seeing "Exceeding usage limits". I would be happy if someone can explain the difference.

Also, I would like to mention that if there was an option to increase the database size by paying a few euros, I would be happy to buy that. I just don't see a reason to upgrade to the Pro plan right now since I don't use any other feature. I hope supabase considers this option in the future.

**Comment by kevcodez** [comment, marked as answer]
Hi @mertbulan

we're currently working on completing the migration to a new usage pipeline in the background. The usage summary at the top of the [organization usage page](https://supabase.com/dashboard/org/_/usage) should already correctly reflect that new pipeline and show you around ~500MB of database space usage.


The chart displaying the historical daily database space usage is still based on the old usage pipeline that takes a single data point per day (max per day), which can lead to these higher values in your case, instead of hourly snapshotting it and averaging it down properly. Over the course of the next few weeks, we are going to replace all these charts to be based on the new usage pipeline, which should fix this once and for all.

**Reply by mertbulan** [reply]
@kevcodez thank you for the fix! I can confirm that I see the right value on the usage page now.

<img width="576" alt="Screenshot 2024-10-08 at 12 01 45" src="https://github.com/user-attachments/assets/5e9abe52-c5c3-464e-8cef-8276e3c75a8b">

**Reply by kevcodez** [reply]
Awesome, thanks for confirming!
